components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/KubernetesGatewayImpl.java

```diff
@@ -0,0 +1,340 @@
+
+        System.setProperty(TRY_KUBE_CONFIG, "false");
+        System.setProperty(TRY_SERVICE_ACCOUNT, "true");
+        ConfigBuilder configBuilder;
+
+        if (masterURL != null) {
+            configBuilder = new ConfigBuilder().withMasterUrl(masterURL);
+        } else {
+            throw new ContainerBasedGatewayException("Kubernetes Master URL is not provided!", ExceptionCodes
+                    .ERROR_INITIALIZING_DEDICATED_CONTAINER_BASED_GATEWAY);
+        }
+
+        if (!StringUtils.isEmpty(saTokenFileName)) {
+            configBuilder.withOauthToken(resolveToken("encrypted" + saTokenFileName));
+        }
+
+        return configBuilder.build();
+    }
+
+    /**
+     * Set Default Openshift client
+     *
+     * @param openShiftClient Openshift client
+     */
+    void setClient(OpenShiftClient openShiftClient) {
+        this.client = openShiftClient;
+    }
+
+    /**
+     * Get resources from template
+     *
+     * @param template Template as a String
+     * @return HasMetadata
+     * @throws ContainerBasedGatewayException if failed to load resource from the template
+     */
+    private HasMetadata getResourcesFromTemplate(String template) throws ContainerBasedGatewayException {
+
+        List<HasMetadata> resources;
+
+        try (InputStream inputStream = IOUtils.toInputStream(template)) {
+
+            resources = client.load(inputStream).get();
+            if (resources == null || resources.isEmpty()) {
+                throw new ContainerBasedGatewayException("No resources loaded from the definition provided : ",
+                        ExceptionCodes.NO_RESOURCE_LOADED_FROM_DEFINITION);
+            }
+            return resources.get(0);
+
+        } catch (IOException e) {
+            throw new ContainerBasedGatewayException("Client cannot load any resource from the template :  "
+                    + template, e, ExceptionCodes.TEMPLATE_LOAD_EXCEPTION);
+        }
+    }
+
+    /**
+     * Create a service in cms
+     *
+     * @param serviceTemplate Service template as a String
+     * @param serviceName     Name of the service
+     * @throws ContainerBasedGatewayException if failed to create a service
+     */
+    private void createServiceResource(String serviceTemplate, String serviceName)
+            throws ContainerBasedGatewayException {
+
+        HasMetadata resource = getResourcesFromTemplate(serviceTemplate);
+
+        try {
+            if (resource instanceof Service) {
+                // check whether there are existing service already
+                if (client.services().inNamespace(namespace).withName(serviceName).get() == null) {
+                    Service service = (Service) resource;
+                    Service result = client.services().inNamespace(namespace).create(service);
+                    if (log.isDebugEnabled()) {
+                        log.debug("CMS Type: " + cmsType + ". The Service Definition : " + serviceTemplate);
```

Comment:
can we follow same logging format where we use {} and list of key words which will be the standradr of our logging